jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -302,15 +302,25 @@ else if (filled < 0)
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("{} caught exception {}", this, _channel.getState(), x);
-            BufferUtil.clear(_requestBuffer);
-            releaseRequestBuffer();
+            if (_retainableByteBuffer != null)
+            {
+                _retainableByteBuffer.clear();
+                try
+                {
+                    releaseRequestBuffer();
+                }
+                catch (Throwable t)
+                {
+                    x.addSuppressed(t);
+                }
+            }
             getEndPoint().close(x);
```

Comment:
I'm nitpicking here, but if we surround this call, then we should surround also all the others -- not sure we want to do that.

How about this:

```suggestion
            try
            {
                if (LOG.isDebugEnabled())
                    LOG.debug("{} caught exception {}", this, _channel.getState(), x);
                if (_retainableByteBuffer != null)
                {
                    _retainableByteBuffer.clear();
                    releaseRequestBuffer();
                }
            }
            finally
            {
                getEndPoint().close(x);
            }
```

Reply:
Okay.